src/containers.c

```diff
@@ -0,0 +1,314 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Philippe Antoine <p.antoine@catenacyber.fr>
+ */
+
+#include "suricata-common.h"
+#include "containers.h"
+#include "util-misc.h"        //ParseSizeStringU64
+#include "util-thash.h"       //HashTable
+#include "util-memcmp.h"      //SCBufferCmp
+#include "util-hash-string.h" //StringHashDjb2
+#include "util-validate.h"    //DEBUG_VALIDATE_BUG_ON
+
+typedef struct ContainerList {
+    THashTableContext *ht;
+    uint32_t timeout;
+} ContainerList;
```

Comment:
this name is confusing. It's a hash table, not a list

Reply:
ok